Civi/Webshoptools/HandleWebShopActivity.php

```diff
@@ -0,0 +1,165 @@
+    ]);
+
+    foreach ($customFields['values'] as $customField) {
+      $data = [];
+      $data['custom_field_data'] = $customField;
+      $data['custom_field_id'] = $customField['id'];
+      $data['entity_field_name'] = 'custom_' . $customField['id'];
+      $data['old_value'] = empty($currentActivity[$data['entity_field_name']]) ? '' : $currentActivity[$data['entity_field_name']];
+
+      if (empty($params['custom'][$customField['id']])) {
+        $data['new_value'] = '';
+        $data['new_value_label'] = '';
+      } else {
+        foreach ($params['custom'][$customField['id']] as $field) {
+          if (!empty($field['id'])) {
+            $fieldRowId = $field['id'];
+          }
+          $data['new_value'] = $field['value'];
+
+          if (!empty($customField['option_group_id']) && !empty($data['new_value'])) {
+            $data['new_value_label'] = civicrm_api3('OptionValue', 'getvalue', [
+              'return' => 'label',
+              'option_group_id' => $customField['option_group_id'],
+              'value' => $data['new_value'],
+            ]);
+          } else {
+            $data['new_value_label'] = $field['value'];
+          }
+        }
+      }
+
+      $preparedCustomFields[$customField['name']] = $data;
+    }
+
+    foreach ($customFields['values'] as $customField) {
+      $preparedCustomFields[$customField['name']]['params_key'] = $preparedCustomFields[$customField['name']]['entity_field_name'] . '_' . $fieldRowId;
+      $preparedCustomFields[$customField['name']]['field_row_id'] = (int) $fieldRowId;
+    }
+
+    return $preparedCustomFields;
+  }
+
+  /**
+   * @return string
+   */
+  private static function generateSubject($customFieldsData) {
+    $orderType = $shirtType = $shirtSize = $orderCount = '';
+
+    if (!empty($customFieldsData['order_type']['new_value_label'])) {
+      $orderType = $customFieldsData['order_type']['new_value_label'] . ' ';
+    }
+
+    if (!empty($customFieldsData['shirt_type']['new_value_label']) && !empty($customFieldsData['shirt_size']['new_value_label'])) {
+      $shirtType = $customFieldsData['shirt_type']['new_value_label'] . '/';
+    } elseif (!empty($customFieldsData['shirt_type']['new_value_label'])) {
+      $shirtType = $customFieldsData['shirt_type']['new_value_label'] . ' ';
+    }
+
+    if (!empty($customFieldsData['shirt_size']['new_value_label'])) {
+      $shirtSize = $customFieldsData['shirt_size']['new_value_label'] . ' ';
+    }
+
+    if (!empty($customFieldsData['order_count']['new_value_label'])) {
+      $orderCount = $customFieldsData['order_count']['new_value_label'] . 'x';
+    }
+
+    return trim($orderType . $shirtType . $shirtSize . $orderCount);
+  }
+
+  /**
+   * @param $currentActivity
+   * @param $params
+   * @param $operation
+   * @return bool
+   */
+  private static function isNewActivityStatusCompleted($currentActivity, $params, $operation) {
+    if ($operation != 'edit') {
```

Comment:
We had some internal discussion on this - we decided we want the same logic to apply when creating *new* activities if the status is set to "Completed". In these cases, we also want to set the exported field + date.

Note that both `$params['status_id'] == \CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', 'Completed');` and `empty($params['status_id'])` could indicate status "Completed" (the default). However, for `empty($params['status_id'])`, this would only be true in the `$operation == 'create'` scenario, because existing activities might already have a different status.

Reply:
This scenario does not work yet (exported is not set):

> Note that both $params['status_id'] == \CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_status_id', 'Completed'); and empty($params['status_id']) could indicate status "Completed" (the default). However, for empty($params['status_id']), this would only be true in the $operation == 'create' scenario, because existing activities might already have a different status.

Example API request:
```cv api4 Activity.create +v 'activity_type_id:name=Webshop Order' +v webshop_information.order_type:name=T_Shirt +v target_contact_id=206 +v source_contact_id=user_contact_id +v webshop_information.order_count=1 +v status_id:name=Completed```